streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -1211,7 +1213,11 @@ void retryWithBackoff(final Map<TaskId, Set<TopicPartition>> tasksToBeCreated) {
                         it.remove();
                     } catch (final LockException e) {
                         // ignore and retry
-                        log.warn("Could not create task {}. Will retry.", taskId, e);
+                        if (log.isDebugEnabled() || log.isTraceEnabled()) {
+                            log.warn("Could not create task {}. Will retry: {}", taskId, e);
```

Comment:
I think, the message should always be WARN level -- but for debug/trace it contains the whole stack trace, while in regular warn mode it only should the error message itself. Open to change it -- only want to explain my thinking behind it.